redux-excersice/src/components/FormComponent.js

```diff
@@ -1,8 +1,66 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
+import { object, string, ref } from 'yup';
 
 export default function FormComponent() {
 
-  const handleSubmit = event => {}
+  const [values, setValues] = useState({
+    name: '',
+    password: '',
+    confirm: '',
+  });
+
+  const [errors, setErrors] = useState({
+    name: false,
+    password: false,
+    confirm: false,
+  })
+
+  const formSchema = object().shape({
+    name: string().required(),
+    password: string().required(),
+    confirm: string().oneOf([ref('password'), null], 'Passwords must match').required(),
+  });
+
+  const onFieldChange = useCallback((event) => {
+    setValues(
+    {
+      ...values,
+      [event.target.name]: event.target.value,
+  
+    }
+  )
+  }, [values]);
+
+
+  const handleSubmit = async (event) => {
+      event.preventDefault()
+
+      const isFormValid = await formSchema.isValid(values, {
+        abortEarly: false, // Prevent aborting validation after first error
+      });
+
+      if (isFormValid) {
+        // If form is valid, continue submission.
+        console.log('Pass')
```

Comment:
We could dispatch a formValid action and then from the parent component do something.